FinalProject/gs-rest-service-master/complete/src/main/java/dao/imp/CartDaoImp.java

```diff
@@ -0,0 +1,182 @@
+package dao.imp;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.springframework.transaction.annotation.Transactional;
+
+import bo.Cart;
+import bo.CartItem;
+import bo.Product;
+import bo.User;
+import connector.DbConnector;
+import dao.CartDao;
+
+public class CartDaoImp implements CartDao {
+
+	@Transactional
+	public int getLastId() throws SQLException {
+		Connection conn = DbConnector.getInstance().getConnection();
+		String query = "SELECT MAX(c.cartId) maxId FROM carts c";
+		ResultSet result;
+		int cartId = 0;
+
+		try {
+			PreparedStatement preparedStatement = conn.prepareStatement(query);

```

Comment:
Minor: You could avoid line 23 with  DbConnector.getInstance().getConnection().prepareStatement(query);